HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+    assuming a root of
+    `target/assets/app/resources/com/redhat/rhjmc/containerjfr/net/web/`.
+
+    ###### response
+
+    `200` - The body is the requested asset. Note that if the requested asset
+    does not exist or the path is invalid, the request will not be handled here
+    and instead  will be routed to
+    [`WebClientAssetsGetHandler`](#WebClientAssetsGetHandler).
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/images/pfbg_992@2x.jpg --output pfbg_992@2x.jpg
+      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                     Dload  Upload   Total   Spent    Left  Speed
+    100  574k  100  574k    0     0  43.1M      0 --:--:-- --:--:-- --:--:-- 43.1M
+    ```
+
+
+* #### `TargetsGetHandler`
+
+    ###### synopsis
+    Scans for and returns discoverable target JVMs.
+
+    ###### request
+    `GET /api/v1/targets`
+
+    ###### response
+    `200` - The body consists of a comma-seperated list of targets,
+    enclosed in square brackets.
+
+    The format for a target is
+    `{"connectUrl":"$CONNECT_URL","alias":"$ALIAS"}`.
+
+    So a list with no recordings looks like `[]`,
+    a list with one recording looks like `[{...}]`,
+    and a list with two recordings looks like `[{...},{...}]`.
+
+    `401` - User authentication failed. The body is an error message.
+
+    `407` - JMX authentication failed. The body is an error message.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/api/v1/targets
+    [{"connectUrl":"service:jmx:rmi:///jndi/rmi://container-jfr:9091/jmxrmi","alias":"com.redhat.rhjmc.containerjfr.ContainerJfr"}]
+    ```
+
+
+* #### `WebClientAssetsGetHandler`
+
+    ###### synopsis
+    Handles all `GET` requests to paths not covered by another handler;
+    returns a simple `index.html`.
+
+    ###### request
+    `GET /*`
+
+    ###### response
+    `200` - The body is the Container JFR web client's
+    `index.html` HTML document.
+
+    `500` - There was an unexpected error. The body is an error message.
+
+    ###### example
+    ```
+    $ curl localhost:8181/foo
+    <!doctype html><html lang="en-US"><head><meta charset="utf-8"><title>ContainerJFR</title><meta id="appName" name="application-name" content="ContainerJFR"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" type="image/png" href="https://www.patternfly.org/components/patternfly/dist/img/favicon.ico"><base href="/"><link href="app.css" rel="stylesheet"></head><body><noscript>Enabling JavaScript is required to run this app.</noscript><div id="root"></div><script src="app.bundle.js"></script></body></html>
+    ```
+
+
+### Flight Recorder
+
+* #### `RecordingDeleteHandler`
+
+    ###### synopsis
+    Deletes a recording that was saved to persistent storage.
```

Comment:
We use the term "archive" in some other places, particularly in the web-client, so perhaps it should also be included here alongside the more technical "persistent storage"?

Reply:
Yeah I wasn't really sure which to use, since I saw "persistent storage" in COMMANDS.md but "archive" elsewhere.
But on second thought, I see that "archive" is the word that's used in the README as well, so maybe we should just skip "persistent storage" altogether? Since anyone wondering about "archive" would be able to find what it is easily that way, and keeping the language consistent with the README is probably best.

Reply:
Sure, using only "archive" sounds fine too.